### ISSUE TITLE
Dataset

### ISSUE THREAD

**Issue by wwy8** [heqin-zhu/BPfold#6]
Hi Heqin,

Thank you for your excellent work. I'm new to RNA structure prediction and I've read your paper – it's really impressive.

I have a couple of questions regarding the dataset:

In the paper, RNAStrAlign_bpseq is described as the training set, but the data downloaded from GitHub appears to be split as a test set. Is my understanding correct?

The paper states that RNAStrAlign_bpseq contains 19,313 samples, but the downloaded data from GitHub has 37,052 samples. Could you please clarify if I’ve missed something?

Does the shared data correspond to the raw data or the preprocessed version used in your experiments?

Thank you very much for your time and help!

<img width="948" height="423" alt="Image" src="https://github.com/user-attachments/assets/2bc0240d-3ec6-468e-9a69-78e918bcf4a3" />
<img width="1053" height="689" alt="Image" src="https://github.com/user-attachments/assets/d5c60ec9-5a6f-4a67-ada1-84a118008084" />

<img width="420" height="567" alt="Image" src="https://github.com/user-attachments/assets/c0dff8c9-8b05-45ea-b746-8c008c105668" />

**Comment by heqin-zhu** [comment]
Hi Wenyan,

Thank you for your interest!

The `RNAStrAlign_bpseq` folder contains the full raw dataset (37,052 seqs). We used a filtered (<=600nt, base filtering, etc.) subset for training, which are indexed in the `data_index.yaml` file (19,313 seqs). The shared data are raw bpseq files; our code handles all preprocessing automatically.

**Comment by wwy8** [comment]
Hi Heqin,
Thank you for your reply. 
In data_index.yaml, I noticed that RNAStrAlign_bpseq is assigned to the test set. However, the paper mentions that RNAStrAlign_bpseq is used for training. Am I misunderstanding something, or is there an issue with the data_index.yaml file? Also, does the data_index.yaml use the train / test fields to distinguish between the training set and the test set? Thank you!

<img width="1278" height="969" alt="Image" src="https://github.com/user-attachments/assets/8f2860fd-870a-4408-ac15-cdff347ce704" />

**Comment by heqin-zhu** [comment]
Hi Wenyan,
Sorry for the confusion. That was my mistake. In the `data_index.yaml` file, I incorrectly placed `RNAStrAlign_bpseq` in the test set; it should be in the training set. The index file defines the train/test splits, and the data_loader will automatically filter out any sequences longer than 600 nt.

**Comment by wwy8** [comment]
Hi Heqin,
Thank you for your reply.
In the RNAStrAlign_bpseq  dataset you kindly shared on GitHub, the number of sequences with a length of 30–600 appears to be 25,981. However, in your paper, the number is stated as 19,313.
May I respectfully ask whether the data shared on GitHub has already undergone redundancy removal and filtering of invalid secondary structures? If not, would it be possible for you to kindly share the post‑processed version of the RNAStrAlign dataset that was used in your paper?
Thank you very much for your time and consideration.

**Comment by heqin-zhu** [comment]
Hi Wenyan,

Thanks for noticing the dataset size difference!
The 19,313 number in the paper refers to a filtered subset I used for training, where I removed duplicate sequences and those with very few base pairs to improve training quality.
Unfortunately, I don't have that exact filtered version saved separately anymore, only the 25,981-sequence dataset is available on server and GitHub.
For comparison, you can definitely use the full dataset or a subset by applying similar filtering yourself if you'd like to ignore noisy annotations. Either way should work fine for benchmarking.
